R/pool-methods.R

```diff
@@ -0,0 +1,89 @@
+#' @include pool.R
+NULL
+
+#' S4 class for compatibility with DBI methods
+#' @export
+setClass("Pool")
+
+## documented manually, with the Pool object
+#' @export
+poolCreate <- function(factory, ...,
+                       minSize = 1, maxSize = Inf,
+                       idleTimeout = 60000, validateTimeout = 2000,
+                       stateEnv = NULL) {
+  pool <- Pool$new(
+    function() {factory(...)},
+    minSize, maxSize,
+    idleTimeout, validateTimeout, stateEnv
+  )
+  ## if a createPool is used with a DBIDriver, make a note
+  ## (to ease dplyr compatibility later on)
+  checkDriver(pool, ...)
```

Comment:
We can certainly store this in the `stateEnv` instead (we didn't have that at the time I wrote this). Why would that be preferable? Just because that's the sort of thing we create `stateEnv` for?

As for where this belongs, that is a decision we must make. I decided to keep it here, so that you can use `poolCreate` directly (instead of `dbPool`) and still have compatibility with `dplyr`. I figured that was more important than the "elegance" factor of keeping this function absolutely generic (it still is though: if there is no `drv` argument, `checkDriver` does nothing). I was thinking, for example, of the situation in which you might have your own variation of `dbConnect` and you pass that in as the `factory` arg (or even if you're just passing in `dbConnect`, do we really want to tell people that they can't use `poolCreate` directly? I guess that would be fine, we haven't really had this discussion yet though...)